stackrox/stackrox
# ROX-11070: Handle ID of groups within UI.

## Description

This is the third PR in a series of PRs regarding removing the composite key from the groups datastore, allowing users to specify different roles for the same group properties.

For full context, you can navigate to the previous PRs by using the comment created below.

This PR will update the UI to include the ID of the default role when creating auth providers, as well as update the group object to include the ID within the properties.

## Testing Performed
- see CI (existing tests regarding auth providers should still pass).

Manual testing
1. Create an auth provider, use minimum access role `None`.
![Screenshot 2022-06-15 at 01 45 30](https://user-images.githubusercontent.com/89904305/173707408-d6ff674b-e660-4a62-a14e-0988a3fd8728.png)
1. Configure multiple rules with the same attribute key and values, but different role mappings.
![Screenshot 2022-06-15 at 01 55 22](https://user-images.githubusercontent.com/89904305/173708374-be2801d9-d510-44b7-81fa-218e17e454bf.png)
1. Login with the configured user and observe multiple roles being assigned.
![Screenshot 2022-06-15 at 01 56 13](https://user-images.githubusercontent.com/89904305/173708452-5fecf2a4-4a40-45c1-8bc2-35d476b9c767.png)
